### PR TITLE
CB-335 - Cluster start is possible for CM

### PR DESCRIPTION
**Pull request by pdarvasi**

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/clouderamanager/ClouderaManagerModificationService.java [hunk 1]
```diff
@@ -80,6 +82,53 @@ public void stopCluster(Stack stack) throws CloudbreakException {
 
     @Override
     public int startCluster(Stack stack) throws CloudbreakException {
```
Comment (gregito) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/clouderamanager/ClouderaManagerModificationService.java]: Do we have unit test for this method? If not, please write.

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/clouderamanager/ClouderaManagerModificationService.java [hunk 2]
```diff
@@ -80,6 +82,53 @@ public void stopCluster(Stack stack) throws CloudbreakException {
...
+            startClouderaManager(stack, client);
+            startAgents(stack, client);
+            return startServices(stack, client).getId().intValue();
+        } catch (ApiException e) {
+            LOGGER.info("Couldn't start Cloudera Manager services", e);
+            throw new AmbariOperationFailedException(e.getMessage(), e);
+        }
+    }
+
+    private ApiCommand startServices(Stack stack, ApiClient client) throws ApiException, CloudbreakException {
```
Comment (gregito) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/clouderamanager/ClouderaManagerModificationService.java]: Do we have unit test for this functionality? If not, please write.

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/clouderamanager/ClouderaManagerModificationService.java [hunk 3]
```diff
@@ -80,6 +82,53 @@ public void stopCluster(Stack stack) throws CloudbreakException {
...
+        } else if (isTimeout(pollingResult)) {
+            throw new CloudbreakException("Timeout while stopping Cloudera Manager services.");
+        }
+        eventService
+                .fireCloudbreakEvent(stack.getId(), UPDATE_IN_PROGRESS.name(),
+                        cloudbreakMessagesService.getMessage(AMBARI_CLUSTER_SERVICES_STARTED.code()));
+        return apiCommand;
+    }
+
+    private void startAgents(Stack stack, ApiClient client) {
```
Comment (gregito) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/clouderamanager/ClouderaManagerModificationService.java]: Do we have unit test for this functionality? If not, please write.

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/clouderamanager/ClouderaManagerModificationService.java [hunk 4]
```diff
@@ -80,6 +82,53 @@ public void stopCluster(Stack stack) throws CloudbreakException {
...
+        LOGGER.debug("Starting Cloudera Manager agents on the hosts.");
+        PollingResult hostsJoinedResult = clouderaManagerPollingServiceProvider.hostsPollingService(stack, client);
+        if (isExited(hostsJoinedResult)) {
+            throw new CancellationException("Cluster was terminated while starting Cloudera Manager agents.");
+        }
+    }
+
+    private void startClouderaManager(Stack stack, ApiClient client) throws CloudbreakException {
```
Comment (gregito) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/clouderamanager/ClouderaManagerModificationService.java]: Do we have unit test for this functionality? If not, please write.

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/clouderamanager/ClouderaManagerModificationService.java (hunk 1 above)
Comment (keyki) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/clouderamanager/ClouderaManagerModificationService.java]: We're covering this with IT mock tests.

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/clouderamanager/ClouderaManagerModificationService.java (hunk 2 above)
Comment (keyki) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/clouderamanager/ClouderaManagerModificationService.java]: Same for every tests

**Pull request merged**